src/Workspaces/Core/Portable/NamingStyles/EditorConfig/EditorConfigNamingStyleParser.cs

```diff
@@ -127,7 +127,11 @@ public static NamingStylePreferences ParseDictionary(IReadOnlyDictionary<string,
 
         private static Dictionary<string, string> TrimDictionary(IReadOnlyDictionary<string, string> allRawConventions)
         {
-            var trimmedDictionary = new Dictionary<string, string>(allRawConventions.Count);
+            // At this point, keys have been lowercased. So, for example, to make a naming style
```

Comment:
@jasonmalinowski It looks like the keys were lowercased long before we got to this feature. Is that expected, or is there a bug deeper in how document options are constructed, or how keys are read initially?

Reply:
why are keys being lowercased?

Reply:
from https://editorconfig.org 
> Currently all properties and values are case-insensitive. They are lowercased when parsed.

Reply:
So I'm not sure where they're being lowercased, but:

https://github.com/dotnet/roslyn/blob/2f57d11ecc36d11c06fc862dcb0de37c96d305d2/src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerConfigOptions.cs#L12-L16

That should be the comparer used in any dictionaries being created. It looks like this change dropped that comparer.

Reply:
The .editorconfig spec does also specifically [say](https://editorconfig.org/#file-format-details):

> Currently all properties and values are case-insensitive. They are lowercased when parsed.

Reply:
Ah here you go:

https://github.com/dotnet/roslyn/blob/3892c636a33f7c3f2e27bc596ec6c43168b57854/src/Compilers/Core/Portable/CommandLine/AnalyzerConfig.cs#L167

Reply:
> Currently all properties and values are case-insensitive. They are lowercased when parsed.

We kinda depend on the values not being lowercased. 😬